pkg/controller/imagepusher/imagepusher_test.go

```diff
@@ -0,0 +1,296 @@
+			}
+		})
+	}
+}
+
+func init() {
+	if err := imagev1.Install(scheme.Scheme); err != nil {
+		panic(fmt.Sprintf("failed to register imagev1 scheme: %v", err))
+	}
+}
+
+func TestReconcile(t *testing.T) {
+	t.Parallel()
+
+	now := metav1.Now()
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   "ci",
+			Name:        "registry-pull-credentials",
+			Annotations: map[string]string{"a": "c"},
+		},
+		Data: map[string][]byte{"pass": []byte("some")},
+		Type: corev1.SecretTypeDockerConfigJson,
+	}
+
+	release46ISTag := &imagev1.ImageStreamTag{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ocp",
+			Name:      "release:4.6",
+		},
+		Image: imagev1.Image{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "sha256:14a6cf5a0b9ef37ca45c3ce75ff0c16d44ddc5e446aa00aec5e7819157e838be",
+				CreationTimestamp: now,
+			},
+			DockerImageReference: "image-registry.openshift-image-registry.svc:5000/ocp/release@sha256:14a6cf5a0b9ef37ca45c3ce75ff0c16d44ddc5e446aa00aec5e7819157e838be",
+		},
+	}
+
+	release46ISTagAPICI := &imagev1.ImageStreamTag{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ocp",
+			Name:      "release:4.6",
+		},
+		Image: imagev1.Image{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "sha256:14a6cf5a0b9ef37ca45c3ce75ff0c16d44ddc5e446aa00aec5e7819157e838be",
+				CreationTimestamp: now,
+			},
+			DockerImageReference: "registry.ci.openshift.org/ocp/release@sha256:14a6cf5a0b9ef37ca45c3ce75ff0c16d44ddc5e446aa00aec5e7819157e838be",
+		},
+	}
+
+	releaseIS := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ocp",
+			Name:      "release",
+			Annotations: map[string]string{
+				"a": "b",
+			},
+		},
+	}
+
+	ctx := context.Background()
+
+	for _, tc := range []struct {
+		name                 string
+		request              types.NamespacedName
+		apiCIClient          ctrlruntimeclient.Client
+		appCIClient          ctrlruntimeclient.Client
+		expected             error
+		expectedAPICIObjects []runtime.Object
+		expectedAPPCIObjects []runtime.Object
+		verify               func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error
+	}{
+		{
+			name: "a new tag is created in app.ci",
+			request: types.NamespacedName{
+				Name:      "release:4.6",
+				Namespace: "ocp",
+			},
+			apiCIClient: bcc(fakeclient.NewFakeClient(secret.DeepCopy())),
+			appCIClient: fakeclient.NewFakeClient(release46ISTag.DeepCopy(), releaseIS.DeepCopy()),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				actualImageStreamImport := &imagev1.ImageStreamImport{}
+				if err := apiCIClient.Get(ctx, types.NamespacedName{Name: "release", Namespace: "ocp"}, actualImageStreamImport); err != nil {
+					return fmt.Errorf("faile to get ImageStreamImport from api.ci: %w", err)
+				}
+				expectedImageStreamImport := &imagev1.ImageStreamImport{
+					TypeMeta: metav1.TypeMeta{
```

Comment:
In a different PR I added a helper to avoid comparing TypeMeta and ResourceVersion, it might be worth using that: https://github.com/openshift/ci-tools/pull/1612/files#diff-9973fc8d4b1a76840c74aba666e57911e52c28af9c1f48f980971c4bdfda3a5fR159

Reply:
Stupid me. I was doing rebase.
However, your PR has not yet merged.
I will wait.

Reply:
Done.